test/components/builders/test_chat_prompt_builder.py

```diff
@@ -0,0 +1,930 @@
+            variables=["var", "required_var"],
+            required_variables=["required_var"],
+        )
+
+        assert comp.to_dict() == {
+            "type": TYPE,
+            "init_parameters": {
+                "template": [
+                    {"content": [{"text": "text and {var}"}], "role": "user", "meta": {}, "name": None},
+                    {"content": [{"text": "content {required_var}"}], "role": "assistant", "meta": {}, "name": None},
+                ],
+                "variables": ["var", "required_var"],
+                "required_variables": ["required_var"],
+            },
+        }
+
+    def test_from_dict(self):
+        comp = ChatPromptBuilder.from_dict(
+            data={
+                "type": TYPE,
+                "init_parameters": {
+                    "template": [
+                        {"content": [{"text": "text and {var}"}], "role": "user", "meta": {}, "name": None},
+                        {
+                            "content": [{"text": "content {required_var}"}],
+                            "role": "assistant",
+                            "meta": {},
+                            "name": None,
+                        },
+                    ],
+                    "variables": ["var", "required_var"],
+                    "required_variables": ["required_var"],
+                },
+            }
+        )
+
+        assert comp.template == [
+            ChatMessage.from_user("text and {var}"),
+            ChatMessage.from_assistant("content {required_var}"),
+        ]
+        assert comp._variables == ["var", "required_var"]
+        assert comp._required_variables == ["required_var"]
+
+    def test_from_dict_template_none(self):
+        comp = ChatPromptBuilder.from_dict(
+            data={
+                "type": TYPE,
+                "init_parameters": {"template": None},
+            }
+        )
+
+        assert comp.template is None
+        assert comp._variables is None
+        assert comp._required_variables is None
+
+
+# new tests
+class TestChatPromptBuilderWithStrTemplate:
+    def test_init(self):
+        template = """
+        {% message role="user" %}
+        Hello, my name is {{name}}!
+        {% endmessage %}
+        """
+        builder = ChatPromptBuilder(template=template)
+        
+        assert builder.template == template
+        assert builder._variables is None
+        assert builder._required_variables is None
+        assert builder.variables == ["name"]
+
+    def test_init_with_invalid_template(self):
+        template = """
+        {% message role="user" %}
+        Hello, my name is {{name}!
+        {% endmessage %}
+        """
+        with pytest.raises(TemplateSyntaxError):
+            ChatPromptBuilder(template=template)        
+
+    def test_run(self):
+        template = """
+        {% message role="user" %}
+        Hello, my name is {{name}}!
+        {% endmessage %}
```

Comment:
Could we also add a multiline example to double check we preserve newline characters for long messages e.g. 
```python
        {% message role="user" %}
        Hello, my name is {{name}}!
        Second line.
        Third line.
        {% endmessage %}
```

Reply:
added in https://github.com/deepset-ai/haystack-experimental/pull/299/commits/d86bd2692167551ea921b7a6eacf0dcd3c3f280c